frame/referenda/src/types.rs

```diff
@@ -282,21 +249,169 @@ impl<
 #[derive(Clone, Eq, PartialEq, Encode, Decode, TypeInfo, MaxEncodedLen)]
 #[cfg_attr(not(feature = "std"), derive(RuntimeDebug))]
 pub enum Curve {
```

Comment:
I have a feeling that this type might benefit to be relocated to `sp-arithmetic`, especially considering that we could use this type to describe XCM fee structures.

Reply:
Eventually possibly, but it's pretty specific for graphs with axes bound between 0 and 1 and where Y reduces as X increases.